### PR TITLE
Fixes a setting overload (typo)

### PR DESCRIPTION
**Pull request by ignasi35**
`mechanism` is a field of the `service-discovery` config object.

### CODE REVIEW

File: runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala [hunk 1]
```diff
@@ -45,7 +45,7 @@ object GrpcClientSettings {
     val defaultServiceConfig = actorSystem.settings.config.getConfig("akka.grpc.client").getConfig("\"*\"")
       .withValue("service-name", ConfigValueFactory.fromAnyRef(serviceName))
       .withValue("port", ConfigValueFactory.fromAnyRef(defaultPort))
-      .withValue("service-discovery-mechanism", ConfigValueFactory.fromAnyRef(serviceDiscoveryMechanism))
+      .withValue("service-discovery.mechanism", ConfigValueFactory.fromAnyRef(serviceDiscoveryMechanism))
```
Comment (raboof) [inline runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala]: How would `discoverService` be used? Would it make more sense to replace it by something like `connectTo(name: String, discovery: SimpleServiceDiscovery, portName: Option[String], protocol: Option[String])` (and introduce `withServiceDiscovery`, `withServicePortName` and `withServiceProtocol` to implement it safely)?

Comment (ignasi35) [reply runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala]: That is a good idea. I'm fighting that exactly while working on https://github.com/lagom/akka-grpc-lagom-quickstart-scala

Comment (ignasi35) [reply runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala]: `connectTo(name: String, discovery: SimpleServiceDiscovery, portName: Option[String], protocol: Option[String])` looks better than passing a service discovery name around.

**Comment by ignasi35** [comment]
PLEASE BUILD

**Comment by raboof** [comment]
Added #399 for the idea discussed in the comments

**Pull request merged**